terralab/auth_helper.py

```diff
@@ -21,7 +25,48 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def get_access_token_with_browser_open(client_info: OAuth2ClientInfo) -> str:
+def get_or_refresh_access_token(cli_config: CliConfig) -> str:
+    """
+    Check for a valid access token; if one exists, return it.
+
+    Otherwise, check for a refresh token; if one exists, attempt to get and save new tokens, and return the access token.
+
+    If refresh attempt fails or if no refresh token is found, prompt user to login via browser, get and save access and refresh tokens, and return the access token.
+
+    Returns a valid access token"""
+    # note that this load function by default only returns valid tokens
+    access_token = _load_local_token(cli_config.token_file)
+    refresh_token = _load_local_token(cli_config.refresh_file, validate=False)
```

Comment:
it can't be validated. from the [azure b2c tokens documentation](https://learn.microsoft.com/en-us/azure/active-directory-b2c/tokens-overview):
- Refresh tokens are opaque to your application. They're issued by Azure AD B2C and can be inspected and interpreted only by Azure AD B2C. 
- "The only way for your application to know if a refresh token is valid is to attempt to redeem it by making a token request to Azure AD B2C."